qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -514,18 +518,25 @@ def _extract_curves(
     ):
         """Extract curve data from experiment data.
 
-        This method internally populate `self._x_values`, `self._y_values`, `self._y_sigmas`
-        and `self._data_index` with given `experiment_data`.
+        This method internally populate two type of curve data.
+
+        - `self.__raw_data`
+            This is the data directly obtained from the experiment data.
+            You can access to this data with `self._raw_data()`.
+            If multiple series exists, you can optionally specify `name` to filter
+            target series.
+
+        - `self.__prepared_data`
+            This is the formatted data created by averaging y and y sigma values
+            over the same x values followed by pre-processing defined by
+            `self._pre_processing()` method. This can be controlled by subclasses.
+            You can access to this data with `self._prepared_data()`.
+            If multiple series exists, you can optionally specify `name` to filter
+            target series.
```

Comment:
This feels a bit strange. It looks like we are doing `pre_processing` after already averaging y and y sigma values so it looks like the preprocessing is in fact post processing. Do we need better names for the `pre_processing` and `post_processing` functions?

Reply:
It depends on the definition of what is the processing. I think this is curve fit because this module is fitter. Pre-processing is format of samples, and post-processing is format of fit values (analysis result). 

The difference of averaging and pre_processing are
- averaging: This can change number of data points (thus series index mapping should be updated)
- pre_processing: This doesn't change data points, such as smoothing

Maybe renaming would be an option. Any suggestions?

Reply:
So should `post_processing` be called `post_process_fit`?

Reply:
renamed to 
- `pre_processing` > `formad_data`
- `post_processing` > `post_analysis`